frontend/apps/cloudstorage/cloudstorage.lua

```diff
@@ -0,0 +1,381 @@
+    local callbackAdd = function(fields)
+        local cs_settings = self:readSettings()
+        local cs_servers = cs_settings:readSetting("cs_servers") or {}
+        if type == "dropbox" then
+            table.insert(cs_servers,{
+                name = fields[1],
+                password = fields[2],
+                type = "dropbox",
+                url = "/"
+            })
+        elseif type == "ftp" then
+            table.insert(cs_servers,{
+                name = fields[1],
+                address = fields[2],
+                username = fields[3],
+                password = fields[4],
+                type = "ftp",
+                url = "/"
+            })
+        end
+        cs_settings:saveSetting("cs_servers", cs_servers)
+        cs_settings:flush()
+        self:init()
+    end
+    if type == "dropbox" then
+        DropBox:config(nil, callbackAdd)
+    end
+    if type == "ftp" then
+        Ftp:config(nil, callbackAdd)
+    end
+end
+
+function CloudStorage:editCloudServer(item)
+    local callbackEdit = function(updated_config, fields)
+        local cs_settings = self:readSettings()
+        local cs_servers = cs_settings:readSetting("cs_servers") or {}
+        if item.type == "dropbox" then
+            for i, server in ipairs(cs_servers) do
+                if server.name == updated_config.text and server.password == updated_config.password then
+                    server.name = fields[1]
+                    server.password = fields[2]
+                    cs_servers[i] = server
+                    break
+                end
+            end
+        elseif item.type == "ftp" then
+            for i, server in ipairs(cs_servers) do
+                if server.name == updated_config.text and server.address == updated_config.address then
+                    server.name = fields[1]
+                    server.address = fields[2]
+                    server.username = fields[3]
+                    server.password = fields[4]
+                    cs_servers[i] = server
+                    break
+                end
+            end
+        end
+        cs_settings:saveSetting("cs_servers", cs_servers)
+        cs_settings:flush()
+        self:init()
+    end
+    if item.type == "dropbox" then
+        DropBox:config(item, callbackEdit)
+    elseif item.type == "ftp" then
+        Ftp:config(item, callbackEdit)
+    end
+end
+
+function CloudStorage:deleteCloudServer(item)
+    local cs_settings = self:readSettings()
+    local cs_servers = cs_settings:readSetting("cs_servers") or {}
+    if item.type == "dropbox" then
+        for i, server in ipairs(cs_servers) do
+            if server.name == item.text and server.password == item.password then
+                table.remove(cs_servers, i)
+                break
+            end
+        end
+    elseif item.type == "ftp" then
+        for i, server in ipairs(cs_servers) do
+            if server.name == item.text and server.password == item.password then
```

Comment:
dropbox and ftp branches look exactly the same? You probably also want to check `server.type`.